### PR TITLE
16374: Turn off Add SEP option for read-only and CSR supervisor

### PR DESCRIPTION
**Pull request by davidplappert**
|Field|Value|
|-----|-----|
| Redmine Ticket Number | [16374](https://devops.dchbx.org/redmine/issues/16374) |
| App-Dev Road Map # | n/a |
| Start Date | n/a |
| Due Date | n/a  |
| App-Dev Priority | UP-Disqueued |
| Development Story Points | 0.5 |
| Peer Review Story Points | n/a |
| Ticket Author | Sarah Bagge |

### CODE REVIEW

File: spec/controllers/exchanges/hbx_profiles_controller_spec.rb [hunk 1]
```diff
@@ -519,5 +519,33 @@
...
 
+  describe "GET add_sep_form" do
+    let(:person) { FactoryGirl.create(:person, :with_consumer_role, :with_family) }
+    let(:user) { double("user", :person => person, :has_hbx_staff_role? => true) }
+    let(:hbx_staff_role) { FactoryGirl.create(:hbx_staff_role, person: person)}
+    let(:hbx_profile) { FactoryGirl.create(:hbx_profile)}
+    let(:permission_yes) { FactoryGirl.create(:permission, :hbx_staff)}
+    let(:permission_no) { FactoryGirl.create(:permission, :can_add_sep => false)}
+
+    it "should return 403 error for users with can_add_sep set to false" do
```
Comment (billy-ran-away) [inline spec/controllers/exchanges/hbx_profiles_controller_spec.rb]: Good no-permission spec. 👍

File: spec/controllers/exchanges/hbx_profiles_controller_spec.rb [hunk 2]
```diff
@@ -473,7 +473,7 @@
       @params = {:person=>{:pid => person.id, :ssn => invalid_ssn, :dob => valid_dob},:jq_datepicker_ignore_person=>{:dob=> valid_dob}, :format => 'js'}
       xhr :get, :update_dob_ssn, @params
       expect(response).to render_template('edit_enrollment')
-    end 
+    end
```
Comment (billy-ran-away) [inline spec/controllers/exchanges/hbx_profiles_controller_spec.rb]: 🥇

File: app/data_migrations/define_permissions.rb [hunk 3]
```diff
@@ -8,13 +8,13 @@ def initial_hbx
     Permission.where(name: /^hbx/).delete_all
   	Permission.create(name: 'hbx_staff', modify_family: true, modify_employer: true, revert_application: true, list_enrollments: true,
   	  send_broker_agency_message: true, approve_broker: true, approve_ga: true,
-  	  modify_admin_tabs: true, view_admin_tabs: true)
+  	  modify_admin_tabs: true, view_admin_tabs: true, can_complete_resident_application: true, can_add_sep: true)
```
Comment (billy-ran-away) [inline app/data_migrations/define_permissions.rb]: Good migrations.

File: app/data_migrations/define_permissions.rb [hunk 4]
```diff
@@ -8,13 +8,13 @@ def initial_hbx
     Permission.where(name: /^hbx/).delete_all
```
Comment (Kevin-Wei) [inline app/data_migrations/define_permissions.rb]: Careful about running this @arasevic

File: lib/tasks/define_permissions.rake [hunk 5]
```diff
@@ -21,6 +21,11 @@ namespace :permissions do
   DefinePermissions.define_task :hbx_admin_can_complete_resident_application => :environment
 end
 
+namespace :permissions do
+  desc 'hbx admin can add sep'
+  DefinePermissions.define_task :hbx_admin_can_add_sep => :environment
```
Comment (billy-ran-away) [inline lib/tasks/define_permissions.rake]: This should fix the problem Kevin found in the migration.

File: spec/data_migrations/permissions_spec.rb [hunk 6]
```diff
@@ -1,42 +1,22 @@
...
-        permission = FactoryGirl.create(:permission, :hbx_staff)
-        role = FactoryGirl.create(:hbx_staff_role, person: person, subrole: "hbx_staff", permission_id: permission.id)
-        subject.hbx_admin_can_complete_resident_application
-      end
-
-      it "updates can_complete_resident_application to true" do
-       expect(Person.all.count).to eq(1)
-       expect(Person.first.hbx_staff_role.permission.can_complete_resident_application).to be true
-      end
-    end
```
Comment (billy-ran-away) [inline spec/data_migrations/permissions_spec.rb]: Not having to test adding updates makes the spec shorter.

**Pull request merged**